models/WebsiteSetting.php

```diff
@@ -59,21 +64,29 @@ public static function getById(int $id): ?WebsiteSetting
     {
         $cacheKey = 'website_setting_' . $id;
 
-        try {
-            $setting = \Pimcore\Cache\RuntimeCache::get($cacheKey);
-            if (!$setting) {
-                throw new Exception('Website setting in registry is null');
-            }
-        } catch (Exception $e) {
+        if (\Pimcore\Cache\RuntimeCache::isRegistered($cacheKey)) {
             try {
-                $setting = new self();
-                $setting->getDao()->getById($id);
-                \Pimcore\Cache\RuntimeCache::set($cacheKey, $setting);
-            } catch (NotFoundException $e) {
-                return null;
+                return \Pimcore\Cache\RuntimeCache::get($cacheKey);
+            } catch (Exception) {
+                // ignore try to load from db
             }
         }
 
+        $setting = new self();
+
+        try {
+            $setting->getDao()->getById($id);
+        } catch (NotFoundException) {
+            return null;
+        }
+
+        \Pimcore\Cache\RuntimeCache::set($cacheKey, $setting);
+
+        Pimcore::getEventDispatcher()->dispatch(
+            new WebsiteSettingEvent($setting),
+            WebsiteSettingEvents::POST_LOAD
+        );
```

Comment:
We just discovered, that `WebsiteSetting::getById()` is not necessarily called when a website setting is retrieved, because they are cached in `\Pimcore\Config::getWebsiteConfig()`, which is probably the way to get website settings most of the time (or `\Pimcore\Config::getWebsiteConfigValue()`, which calls the former). 
This method loads *all* website settings and caches them, so `WebsiteSetting::getById()` shouldn't be called most of the time, which makes the event quite unreliable, as it will either not be dispatched (when the settings are cached) or dispatched for *all* settings if they're not cached.